src/software/simulated_tests/non_terminating_validation_functions/robots_avoid_ball_validation.cpp

```diff
@@ -19,8 +17,8 @@ void robotsAvoidBall(double min_distance, std::vector<RobotId> excluded_robots,
             std::find(excluded_robots.begin(), excluded_robots.end(), robot.id()) ==
                 excluded_robots.end())
         {
-            FAIL() << "Robot " + std::to_string(robot.id()) + " is less than " +
-                          std::to_string(min_distance) + " m away from the ball!";
+            yield("Robot " + std::to_string(robot.id()) + " is less than " +
```

Comment:
If we remove the FAIL() keyword, wouldn't the simulated test still report the nonterminating function to have passed even if it didn't?

Reply:
If we use the NonterminatingFunctionValidator, then it should catch that an error message is being yielded and ADD_FAILURE with that error message and cause the test to fail